pkg/kubectl/cmd/util/factory.go

```diff
@@ -327,6 +347,7 @@ func (f *factory) FlagSet() *pflag.FlagSet {
 func (f *factory) Object() (meta.RESTMapper, runtime.ObjectTyper) {
 	cfg, err := f.clientConfig.ClientConfig()
 	checkErrWithPrefix("failed to get client config: ", err)
+	updateAuthFromPrompt(f.flags, cfg)
```

Comment:
Making the change like this means that you'll end up chasing every spot where a clientconfig is requested.